### PR TITLE
Fix package url 

### PR DESCRIPTION
**Pull request by mfoti**

### CODE REVIEW

File: lib/publish.js [hunk 1]
```diff
@@ -102,8 +102,9 @@ export default async (pluginConfig, context) => {
               throw error;
             }
 
-            const { url } = response.file;
-
+            //const { url } = response.file;
```
Comment (JonasSchubert) [inline lib/publish.js:105]: As this is no longer needed, I would delete this line of code.

File: lib/publish.js [hunk 2]
```diff
@@ -102,8 +102,9 @@ export default async (pluginConfig, context) => {
...
+	    const url = uploadEndpoint;
```
Comment (JonasSchubert) [inline lib/publish.js:106]: The `uploadEndpoint` also includes a [`status` and `select` query](https://github.com/semantic-release/gitlab/blob/master/lib/publish.js#L91-L95) which is not required for the download. I would suggest to create a `genericsEndpoint` which is basically the `downloadEndpoint` and enrich it with the query for the upload. Something like:
```js
const genericsEndpoint= urlJoin(
  gitlabApiUrl,
  `/projects/${encodedRepoId}/packages/generic/release/${encodedVersion}/${encodedLabel}`
);

const uploadEndpoint = `${genericsEndpoint}?${status ? `status=${status}&` : ""}select=package_file`;
```

**Review by JonasSchubert — COMMENTED** [review]
Thanks for your contribution @mfoti. I left some remarks ;)

You will also have to adjust the unit tests. The URL changed and theoretically the unit tests will now fail.
As you suggested I would recommend to squash the commits.
And as discussed in #503 this is IMO another breaking change and should be released together with the mentioned PR.

**Comment by JonasSchubert** [comment]
Hi @mfoti any feedback or anything I can help with?

**Comment by JonasSchubert** [comment]
@fgreinacher as @mfoti seems to be busy right now, shall we wait until they come back or merge their changes into my current PR? what would you suggest?

**Comment by fgreinacher** [comment]
> @fgreinacher as @mfoti seems to be busy right now, shall we wait until they come back or merge their changes into my current PR? what would you suggest? 

@JonasSchubert Feel free to cherry-pick into your PR 👍

**Comment by mfoti** [comment]
yes sorry, busy days at work 😔

**Comment by fgreinacher** [comment]
> yes sorry, busy days at work 😔

No worries, you already did the major work and we can finish it 😀

**Comment by JonasSchubert** [comment]
> > yes sorry, busy days at work 😔
> 
> No worries, you already did the major work and we can finish it 😀

Exactly, thanks @mfoti 😃

**Comment by JonasSchubert** [comment]
@mfoti I cherry-picked your commit [here](https://github.com/semantic-release/gitlab/pull/503/commits/e346b22aac3e4e8629ab5fcb9249d545a5541cd4) and implemented some of my suggestions. Feel free to comment 😄

**Comment by fgreinacher** [comment]
Superseded by #503

**Pull request closed without merging**